internal/deviceplugin/deviceplugin.go

```diff
@@ -170,26 +172,41 @@ func waitForServer(socket string, timeout time.Duration) error {
 }
 
 // MakeAllocateResponse creates response data for Allocate GRPC call
-func MakeAllocateResponse(rqt *pluginapi.AllocateRequest, devices map[string]DeviceInfo) (*pluginapi.AllocateResponse, error) {
+func MakeAllocateResponse(rqt *pluginapi.AllocateRequest, devices map[string]DeviceInfo, pluginPrefix string) (*pluginapi.AllocateResponse, error) {
 	resp := new(pluginapi.AllocateResponse)
 	for _, crqt := range rqt.ContainerRequests {
+		var envmap = make(map[string]string)
 		cresp := new(pluginapi.ContainerAllocateResponse)
-		for _, id := range crqt.DevicesIDs {
+		for devNum, id := range crqt.DevicesIDs {
 			dev, ok := devices[id]
 			if !ok {
 				return nil, fmt.Errorf("Invalid allocation request with non-existing device %s", id)
 			}
 			if dev.State != pluginapi.Healthy {
 				return nil, fmt.Errorf("Invalid allocation request with unhealthy device %s", id)
 			}
+			if pluginPrefix == "intelQAT" {
```

Comment:
I think we need here some more generic solution. This part is common code, it's not good to have plugin specifics here. It will be better to pass envs as argument, so each device plugin can control what kind of envs and mount options are passed to container.

Reply:
@rojkov @bart0sh can you look how to refactor this bit better ?

Reply:
@kad I'm already looking at the PR. I'll send my recommendations later today.

Reply:
Sure, I am open to recommendations for refactoring this. Let me know what you guys think.

Reply:
The function `MakeAllocateResponse()` returns a `pluginapi.AllocateResponse` instance which can be modified in the caller of `MakeAllocateResponse()`. Indeed there's no need to have plugin-specific code here.